scripts/ipkg-build

```diff
@@ -68,6 +68,40 @@ pkg_appears_sane() {
 	return $PKG_ERROR
 }
 
+resolve_file_mode_id() {
```

Comment:
```bash
resolve_file_mode_id() {
  local type=$1
  local name=$2
  local c

  case "$name" in
    root) echo 0 ;;
    *[!0-9]*)
      [ "$type" = group ] && c=":" || c=" "

      id=$(sed -rne '0,/^Require-User:.*'"$c$name"'=/s/^.+'"$c$name"'=([0-9]+)\b.*$/\1/p' $TOPDIR/tmp/.packageinfo)
      [ -n "$id" ] || {
        echo "No $type ID found for $name" >&2
        return 1
      }

      echo "$id"
    ;;
    *) echo "$name" ;;
  esac
}
```

Reply:
I've merged this already. Can you push a follow up commit with these (nice) enhancements?

Reply:
> I've merged this already. Can you push a follow up commit with these (nice) enhancements?

Some more lead time to comment on the PR would have been nice. After looking again at what has been pushed to the tree I must say that the entire code is smelly and the overall approach inefficient.

- Why not extend ./scripts/package-metadata.pl to produce the uid/gid map file in a format suitable for consumption by ipkg-build like it is done for other auxiliary data? 
- Why not break Require-User lines with multiple tuples into distinct lines to simplify parsing in ipkg-build which is invoked way more times than the initial map generation?
- Why not break the user/group tuples into separate uid- and gid-map files to avoid the need for slow and failure-prone shell parsing in ipkg-build?
- Why so many subshells can `cut` invocations for trivial string splitting tasks?

Reply:
tl;dr: I'm not aware of many great features of the build system and neither a shell wizard. 

Thanks for the refactored implementation.